enhancements/sig-architecture/58-addon-placement-annotation/README.md

```diff
@@ -0,0 +1,62 @@
+
+## Summary
+
+The proposal introduces two annotations to help user to specify node selector and tolerations for their addons.
+
+## Motivation
+
+Sometimes user want to place their addons on specific nodes, so we need a way to set the node selector and tolerations for a addon.
+
+## Proposal
+
+We introduce two annotations to set node selector and tolerations for a addon
+
+1. Node selector annotation, the key of this annotation is: `addon.open-cluster-management.io/nodeSelector`, the value of this annotation represents a node selector with json format
+2. Tolerations annotation, the key of this annotation is: `addon.open-cluster-management.io/tolerations`, the value of this annotation represents a toleration array with json format
+
+There is an example
+
+```yaml
+apiVersion: addon.open-cluster-management.io/v1alpha1
+kind: ManagedClusterAddOn
+metadata:
+  name: work-manager
+  namespace: cluster1
+  annotations:
+    addon.open-cluster-management.io/nodeSelector: "{\"kubernetes.io/os\":\"linux\"}"
+    addon.open-cluster-management.io/tolerations: "[{\"key\":\"node.kubernetes.io/unreachable\",\"operator\":\"Exists\",\"effect\":\"NoSchedule\"}]"
+spec:
+  installNamespace: open-cluster-management-agent-addon
+```
+
+### Design Details
+
+Currently, there is a function [`GetValuesFromAddonAnnotation`](https://github.com/open-cluster-management-io/addon-framework/blob/v0.3.0/pkg/addonfactory/helper.go#L21) in the addon-framework, it returns a [`Values`]([https://github.com/open-cluster-management-io/addon-framework/blob/v0.3.0/pkg/addonfactory/addonfactory.go#L22) object (the object type is `map[string]interface{}`) that is built from the addon values annotation, this object is used to render the addon deployment.
+In this proposal, if one addon has the node selector and tolerations annotations, this function will merge them to the `Values` object with the key `NodeSelector` and `Tolerations`.
+
+For a addon implementation, developers use [`AgentAddonFactory`](https://github.com/open-cluster-management-io/addon-framework/blob/v0.3.0/pkg/addonfactory/addonfactory.go#L28) to build their addon, they use [`WithGetValuesFuncs`](https://github.com/open-cluster-management-io/addon-framework/blob/v0.3.0/pkg/addonfactory/addonfactory.go#L61) to specify the get values functions (the function type is (`GetValuesFunc`)[https://github.com/open-cluster-management-io/addon-framework/blob/v0.3.0/pkg/addonfactory/addonfactory.go#L24]) to render their addon deployment, for get values function, the developers can choose to use addon-frameworks `GetValuesFromAddonAnnotation` function or implement it by themselves to get these two annotations.
+
+### Test Plan
+- test add placement annotation to a addon which has no values annotation 
+- test add placement annotation to a addon which has values annotation 
+- test delete placement annotation from a addon
+
+### Graduation Criteria
+N/A
+
+### Upgrade Strategy
+N/A
```

Comment:
any concern on the upgrade?

Reply:
add a probable upgrade case, and expecting other reviewers give more comments